Neutron Core/Scripts/neutron_core/ui_dialog.py

```diff
@@ -0,0 +1,53 @@
+import services
+import traceback
+import os
+from .utils import ParametrizedLocalizedString
+from sims4.localization import LocalizationHelperTuning, _create_localized_string
+from ui.ui_dialog import ButtonType, UiDialog, UiDialogResponse
+
+
+class UiDialogChoices(UiDialog):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._choice_responses = list()
+
+    def _get_responses_gen(self):
+        # yield old values:
+        super_results = list(super()._get_responses_gen())
+        for result in super_results:
+            yield result
+        # yield our choices:
+        for response in self._choice_responses:
+            yield response
+
+
+# Creates, initializes and displays the choice dialog
+def display_choices(choices, choice_callback, text: ParametrizedLocalizedString = None, title: ParametrizedLocalizedString = None, cancel: ParametrizedLocalizedString = None):
```

Comment:
Could I get an example of how this would be used? What could `choices` be, and how should they be handled in `choice_callback`?

Reply:
`choices` is a list of STBL entries. As for the callback, a if/else block or a switch case would do. The callback function itself only needs one parameter which is an integer that will range from 0 to n. n being the number of choices you supplied and in the order you supplied them. The game's default response ID for `cancel` is 10002.

Reply:
The STBL entries would need to be accessed from a .package file, right?

Reply:
They need to be loaded into the game at the very least. The value itself could be loaded through a snippet or hardcoded wthin the code of the mod that implements a menu.